src/core/lib/gpr/useful.h

```diff
@@ -19,48 +19,78 @@
 #ifndef GRPC_CORE_LIB_GPR_USEFUL_H
```

Comment:
Suggest moving this file to the gprpp directory, now that it provides a C++ API.

Reply:
Considering leaving it here for now.

The gprpp/gpr separation has lost it's value by now - most things are in gprpp, and the things that are not are probably bugs we should address.

Instead, I'd propose making src/lib/grpc/support, and moving both directories into it.

Reply:
Oh, sorry, forgot to reply to this: The intent of the division between gpr and gprpp is that gpr is C APIs that are exposed above the surface layer for wrapped languages, but gprpp is C++ APIs used only inside of core.

We used to have a single support directory, and we split it out to try to keep these things separate.

Reply:
Ok... I guess I was confused since we didn't actually do that - much of gpr is not exposed up, but only to grpc, and so I think the separation is only leading to confusion.

Reply:
Right, the idea was to migrate things over time.  Anything not expected to be exposed in the C-core API was going to be rewritten in C++ and moved to gprpp.  So we could tell what was left to migrate by looking for things in gpr that were not exposed in the C-core API.